recipes/opencv/4.x/conanfile.py

```diff
@@ -576,7 +576,7 @@ def add_components(components):
                             self.cpp_info.components[conan_component].libdirs.append("lib")
                             self.cpp_info.components[conan_component].libs += tools.collect_libs(self)
 
-                if self.settings.os == "iOS" or self.settings.os == "Macos":
+                if self.settings.os == "iOS" or self.settings.os == "Macos" or self.settings.os == "Linux":
```

Comment:
```suggestion
                if self.settings.os in ["iOS", "Macos", "Linux"]:
```

Reply:
this hack will break soon or later I'm affraid

Reply:
Why do you think that this will break? Do you have something morerobust in mind?

Reply:
Not yet, it's just a comment for other maintainers, the hack following this condition is dangerous because it can add libs in `opencv_core` component that are already listed in other components (like `correspondence`, `multiview` and `numeric`).